src/build_loop.rs

```diff
@@ -106,23 +105,38 @@ impl<'a> BuildLoop<'a> {
         let paths = reduce_paths(&paths);
         debug!("  -> reduced to: {:?}", paths.len());
 
-        debug!("named drvs: {:#?}", build.named_drvs);
+        debug!("named drvs: {:#?}", build.output_paths);
 
-        let mut event = BuildResults {
-            drvs: HashMap::new(),
-            named_drvs: HashMap::new(),
+        // TODO: once build returns realized drvs, make those into roots
+        // create root for every field in OutputPaths
+        // let output_paths = roots.create_roots(build.output_paths)?;
+        let root_gc = roots.add(
+            // TODO: bad, this magic name is not known here (only in roots)
+            "attr-shell_gc_root",
+            &build
+                .drvs
+                .get(0)
+                .expect("logged-evaluation.nix didn’t return a drv!"),
+        )?;
+        let root_shell = roots.add(
+            // TODO: bad, this magic name is not known here (only in roots)
+            "attr-shell",
+            // TODO: add actual root (this is just the same as the shell_root
+            &build
+                .drvs
+                .get(0)
+                .expect("logged-evaluation.nix didn’t return a drv!"),
+        )?;
```

Comment:
Do we want to be rooting the same drv twice?

Reply:
Temporary, will go away once the `TODO`s are cleaned up.

Reply:
outdated by rebase